x-pack/plugins/security_solution/server/search_strategy/endpoint_fields/index.ts

```diff
@@ -40,25 +40,34 @@ export const eventFiltersFieldsProvider = (
 
   return {
     search: (request, _, deps) =>
-      from(requestEventFiltersFieldsSearch(context, request, deps, beatFields, indexPatterns)),
+      from(requestEndpointFieldsSearch(context, request, deps, beatFields, indexPatterns)),
   };
 };
 
-export const requestEventFiltersFieldsSearch = async (
+export const requestEndpointFieldsSearch = async (
   context: EndpointAppContextService,
   request: IndexFieldsStrategyRequest<'indices'>,
   deps: SearchStrategyDependencies,
   beatFields: BeatFields,
   indexPatterns: DataViewsServerPluginStart
 ): Promise<IndexFieldsStrategyResponse> => {
-  const { canWriteEventFilters } = await context.getEndpointAuthz(deps.request);
+  const { canWriteEventFilters, canReadEndpointList } = await context.getEndpointAuthz(
+    deps.request
+  );
 
-  if (!canWriteEventFilters) {
-    throw new EndpointAuthorizationError();
+  if (
```

Comment:
nice reordering for checking user input first, and then the authorization :fire: